### PR TITLE
WW-133 Hook into SendGridPostBackApiControlle

### PR DESCRIPTION
**Pull request by dianafa**
https://wikia-inc.atlassian.net/browse/WW-133

### CODE REVIEW

**Comment by rybmat** [comment]
LG

File: extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php [hunk 1]
```diff
@@ -103,6 +117,45 @@ public static function onBeforePageDisplay( \OutputPage $out, \Skin $skin ) {
...
+		$dbname = \WikiFactory::IDtoDB( $wikiId );
+		$db = wfGetDB( DB_SLAVE, [ ], $dbname );
+		$revision = Revision::loadFromId( $db, $revisionId );
+
+		if ( $revision ) {
+			$url = self::TRACKING_URL .
+				'?wiki_id=' . $wikiId .
+				'&email_action=' . $action .
+				'&page_id=' . $revision->getTitle()->getArticleID() .
+				'&user_id=' . $revision->getUser();
```
Comment (sebastianmarzjan) [inline extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php:154]: Hmm, do we want to track revision Id too?

Comment (dianafa) [inline extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php:154]: 👍

File: extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php [hunk 2]
```diff
@@ -98,11 +98,10 @@ public static function onPageHistoryBeforeList() {
 
 	public static function onSendGridPostbackLogEvents( $events ) {
 		foreach ( $events as $event ) {
-			if ( $event[ 'event' ] == 'click' && strpos( $event[ 'category' ], self::EMAIL_CATEGORY ) !== false ) {
+			if ( self::checkIfAppreciationEmailEvent( $event ) ) {
```
Comment (sebastianmarzjan) [inline extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php]: self::isAppreciationEmailEvent( $event ) would be shorter and more 'conventional'

**Comment by sebastianmarzjan** [comment]
One suggestion to rename method, rest LGTM

**Pull request merged**